primo-explore/custom/main.js

```diff
@@ -499,6 +499,94 @@ angular.module('myILL', []).component('prmLoansOverviewAfterAppStoreGenerated',
 
 //End My-ill
 
+// Altmetric plugin
+
+app.controller('FullViewAfterController', ['angularLoad', '$http', '$scope', '$element', '$timeout', '$window', function (angularLoad, $http, $scope, $element, $timeout, $window) {
+  var vm = this;
+  this.$http = $http;
+  this.$element = $element;
+  this.$scope = $scope;
+  this.$window = $window;
+  vm.$onInit = function () //wait for all the bindings to be initialised
+  {
+    vm.parentElement = this.$element.parent()[0]; //the prm-full-view container
+    try {
+        vm.doi = vm.parentCtrl.item.pnx.addata.doi[0] || '';
+    } catch (e) {
+      return;
+    }
+
+    if (vm.doi) {
+      //If we've got a doi to work with check whether altmetrics has data for it.
+      //If so, make our div visible and create a new Altmetrics service
+      $timeout(function () {
+        $http.get('https://api.altmetric.com/v1/doi/' + vm.doi).then(function () {
+            try {
+              //Get the altmetrics widget
+              angularLoad.loadScript('https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js?' + Date.now()).then(function () {});
```

Comment:
It might be good to put the url you have here into a constant or variable. I don't know how likely it is to change, so maybe that's unnecessary work.

Reply:
That is a good idea, considering that we are going to try to grab a pubmed id if there is no doi.